src/Qwiq.Core/ReadOnlyObjectWithNameCollection.cs

```diff
@@ -1,33 +1,28 @@
-﻿using System;
-using System.Collections;
+﻿using JetBrains.Annotations;
+using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Diagnostics.Contracts;
 
-using JetBrains.Annotations;
-
 namespace Microsoft.Qwiq
 {
     /// <summary>
     ///     Base class for common operations for Collections.
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    public abstract class ReadOnlyObjectWithNameCollection<T> : IReadOnlyObjectWithNameCollection<T>
+    public abstract class ReadOnlyObjectWithNameCollection<T> : ReadOnlyObjectCollection<T>, IReadOnlyObjectWithNameCollection<T>
+    //TODO: Restrict T to INamed
```

Comment:
The way these types have functionality applied with inheritance creates a problem--this will need to be rethought to permit this restriction. For now the function to define the name can be provided via ctor. This creates duplication among implementors of the class with named objects